### ISSUE TITLE
Exception while handling WantReadError in v1.14.0

### ISSUE THREAD

**Issue by terratrue-daniel** [denisenkom/pytds#158]
Started running into an issue in v1.14.0 that I didn't see in v1.13.0:
```
  File "<redacted>/venv/lib/python3.10/site-packages/pytds/tls.py", line 172, in establish_channel
    conn.do_handshake()
  File "<redacted>/venv/lib/python3.10/site-packages/OpenSSL/SSL.py", line 2202, in do_handshake
    self._raise_ssl_error(self._ssl, result)
  File "<redacted>/venv/lib/python3.10/site-packages/OpenSSL/SSL.py", line 1810, in _raise_ssl_error
    raise WantReadError()
OpenSSL.SSL.WantReadError

During handling of the above exception, another exception occurred:

  ...
  File "<redacted>/venv/lib/python3.10/site-packages/pytds/tls.py", line 190, in establish_channel
    r.begin_response()
  File "<redacted>/venv/lib/python3.10/site-packages/pytds/tds_reader.py", line 198, in begin_response
    raise RuntimeError(
RuntimeError: begin_response was called before previous response was fully consumed
```

I saw that [this line](https://github.com/denisenkom/pytds/blame/master/src/pytds/tls.py#L190) was recently introduced as part of a refactor. But I don't fully understand why it needed to be added.

**Comment by denisenkom** [comment]
Thank you for reporting. I wasn't able to reproduce this issue in my environment.  Can you provide steps to reproduce?

**Comment by terratrue-daniel** [comment]
> Thank you for reporting. I wasn't able to reproduce this issue in my environment. Can you provide steps to reproduce?

I receive this error when I try to connect to a SQL Server instance set up in Azure. `pytds` is being consumed via SQLAlchemy. Let me know if there's anything specific you need to know about the SQL Server instance -- it seems to only affect Azure since I have not encountered this issue when it's deployed in GCP or RDS.

**Comment by tmeckel** [comment]
@denisenkom pinning python-tds to 1.13.0 solved the issue. Must be definitely something with 1.14.0

**Comment by denisenkom** [comment]
I will try testing with Azure

**Comment by denisenkom** [comment]
Fix is ready, pending release of a new version

**Comment by denisenkom** [comment]
Fix released in 1.15.0
